identity/identity.go

```diff
@@ -165,3 +174,62 @@ func NewAuthenticator(authMode AuthMode, httpClient *goosehttp.Client) Authentic
 		return &V3UserPass{client: httpClient}
 	}
 }
+
+type authInformationLink struct {
+	Href string `json:"href"`
+	Rel  string `json:"base"`
+}
+
+type authInformationMediaType struct {
+	Base      string `json:"base"`
+	MediaType string `json:"type"`
+}
+
+type authInformationValue struct {
+	ID         string                     `json:"id"`
+	Links      []authInformationLink      `json:"links"`
+	MediaTypes []authInformationMediaType `json:"media-types"`
+	Status     string                     `json:"status"`
+	Updated    string                     `json:"updates"`
+}
+
+type authInformationVersions struct {
+	Values []authInformationValue `json:"values"`
+}
+
+type authInformation struct {
+	Versions authInformationVersions `json:"versions"`
+}
+
+func FetchAuthOptions(url string, client *goosehttp.Client) (AuthOptions, error) {
```

Comment:
even though goose lacks comments where it should have them, we should comment significant new work like this